R/create_query_data.R

```diff
@@ -0,0 +1,894 @@
+#' Creates queries dataset from SMQ and SDG database or custom definitions as
+#' input dataset for `derive_vars_query()`
+#'
+#' Creates queries dataset from definitions in SMQ and SDG database or custom
+#' definitions as input dataset for `derive_vars_query()`.
+#'
+#' @param queries List of queries
+#'
+#'   A list of `query()` objects is expected.
+#'
+#' @param meddra_version MedDRA version
+#'
+#'   The MedDRA version used for coding the terms in the AE dataset should be
+#'   specified. If any of the queries is a SMQ or a customized query including a
+#'   SMQ, the parameter needs to be specified.
+#'
+#'   *Permitted Values*: A character string
+#'
+#' @param whodd_version WHO Drug Dictionary version
+#'
+#'   The version of the WHO Drug Dictionary used for coding the terms in the CM
+#'   dataset should be specified. If any of the queries is a SDG, the parameter
+#'   needs to be specified.
+#'
+#'   *Permitted Values*: A character string
+#'
+#' @param get_smq_fun Function which returns the terms of a SMQ
+#'
+#'   For each query specified for the `queries` parameter which refers to a SMQ
+#'   the specified function is called to retrieve the terms defining the query.
+#'
+#'   The function must return a dataset with all the terms defining the SMQ. The
+#'   output dataset must contain the following variables.
+#'
+#'   - `TERM_LEVEL`
+#'   - `TERM_NAME` if the variable `TERM_LEVEL` is referring to is character
+#'   - `TERM_ID` if the variable `TERM_LEVEL` is referring to is numeric
+#'   - `QUERY_NAME`
+#'
+#'   The function must provide the following parameters
+#'
+#'   - `smq_select`: A `smq_select()` object.
```

Comment:
Yes, the `smq_select` object is provided by `query.definition` but it does not need to be provided for the `get_smq_fun` parameter. This parameter expects just a function name, e.g., `get_smq_fun = get_smq_terms` but not a call. I hope it will become clearer once I have added example functions to admiral.test and used them in the examples.